db/db_impl.cc

```diff
@@ -912,6 +913,25 @@ Status DBImpl::SetDBOptions(
     InstrumentedMutexLock l(&mutex_);
     s = GetMutableDBOptionsFromStrings(mutable_db_options_, options_map,
                                        &new_options);
+    if (new_options.bytes_per_sync == 0) {
+      new_options.bytes_per_sync = 1024 * 1024;
+    }
+    DBOptions new_db_options =
+        BuildDBOptions(immutable_db_options_, new_options);
+    if (s.ok()) {
```

Comment:
lines 919-920 (maybe lines 916-917 too) should also be inside s.ok(), as there is no reason to call `BuildDBOptions` if `GetMutableDBOptionsFromStrings` itself has failed.

Reply:
It would be less optimized. new_db_options has to be outside s.ok() clause so that it could be reused below. However if we separate the declaration from assignment, the compiler might create two option object: 1st with default values, and 2nd when it is assigned. Assuming at SetOptions mostly pass through, this way is more optimized.

Reply:
that makes sense. Thanks!